go/vt/wrangler/reparent.go

```diff
@@ -604,9 +604,17 @@ func (wr *Wrangler) plannedReparentShardLocked(ctx context.Context, ev *events.R
 
 		rp, err = wr.tmc.PromoteSlaveWhenCaughtUp(promoteCtx, masterElectTabletInfo.Tablet, rp)
 		if err != nil || (ctx.Err() != nil && ctx.Err() == context.DeadlineExceeded) {
+			baseCtx := ctx
+			// It is possible that we have used up the original context,
+			// but at this point we really need to be able to Undo so as not to
+			// leave the cluster in a bad state.
+			// In this case we create a fresh context based on context.Background().
+			if ctx.Err() != nil && ctx.Err() == context.DeadlineExceeded {
```

Comment:
It is not clear to me what this extra check adds in value. I think would just always start from a background context for the undo operation.

Reply:
I was trying to be a miser about creating a new context only if needed, but it actually makes sense to always do it (because you don't want the original context to expire during the call to UndoDemoteMaster).